app/controllers/listing_images_controller.rb

```diff
@@ -75,11 +72,16 @@ def new_image(params, url)
     listing_image.image_downloaded = if url.present? then false else true end
 
     if listing_image.save
-      unless listing_image.image_downloaded
-        listing_image.delay.download_from_url(url)
+      if !listing_image.image_downloaded
+        logger.info("Asynchronously downloading image", :start_async_image_download, listing_image_id: listing_image.id, url: url, params: params)
+        Delayed::Job.enqueue(DownloadListingImageJob.new(listing_image.id, url), priority: 1)
+      else
+        logger.info("Listing image downloaded already", :image_downloaded_already, listing_image_id: listing_image.id, params: params)
       end
+
       render json: ListingImageJSAdapter.new(listing_image).to_json, status: 202, content_type: 'text/plain' # Browsers without XHR fileupload support do not support other dataTypes than text
     else
+      logger.error("Saving listing image failed", :saving_listing_image_failed, params: params)
```

Comment:
Should this log row include something about the errors that prevented image being saved?